### ISSUE TITLE
crash with x11perf

### ISSUE THREAD

**Issue by uli42** [ArcticaProject/nx-libs#640]
nxagent had been locally: `nxagent :55`

```
$ DISPLAY=:55 x11perf -repeat 1 -time 1 -reps 1 -aa4trap1
x11perf - X11 performance program, version 1.2
The X.Org Foundation server version 70000000 on :55
from traube
Sun Jan 21 14:34:01 2018

Sync time adjustment is 0.0000 msecs.
```

Here's the crash:
```
Program received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x0000000000509883 in fbCompositeRect (scanline_buffer=0x7fffffff7660, data=<synthetischer Zeiger>) at fbcompose.c:3493
#2  fbCompositeGeneral (op=op@entry=3 '\003', pSrc=pSrc@entry=0xe10d70, pMask=pMask@entry=0xe10710, pDst=pDst@entry=0xe10ef0, xSrc=0, ySrc=<optimized out>, xMask=0, yMask=0, xDst=3, yDst=3, width=600, height=600) at fbcompose.c:3609
#3  0x00000000004e6d5c in fbComposite (op=3 '\003', pSrc=0xe10d70, pMask=0xe10710, pDst=0xe10ef0, xSrc=<optimized out>, ySrc=<optimized out>, xMask=<optimized out>, yMask=<optimized out>, xDst=3, yDst=<optimized out>, width=600, height=600) at fbpict.c:1120
#4  0x000000000044cc43 in damageComposite (op=<optimized out>, pSrc=0xe10d70, pMask=0xe10710, pDst=0xe10ef0, xSrc=<optimized out>, ySrc=<optimized out>, xMask=0, yMask=0, xDst=0, yDst=0, width=600, height=600) at ../../miext/damage/damage.c:546
#5  0x000000000043a900 in ProcRenderComposite (client=0xdfed10) at NXrender.c:586
#6  0x00000000004352b9 in ProcRenderDispatch (client=<optimized out>) at NXrender.c:1685
#7  0x00000000004349a6 in Dispatch () at NXdispatch.c:482
#8  0x000000000040f018 in main (argc=2, argv=0x7fffffffda78, envp=<optimized out>) at main.c:355
```

**Comment by Ionic** [comment]
Reopening until #645, #648 and #650 have been merged (as per the comment in the obsolete #641 PR).

**Comment by uli42** [comment]
Why are you re-opening? the fb fix is sufficient for the x11perf issue, I have tested that!

**Comment by Ionic** [comment]
Because, in, #641, you explicitly said

> This fixes #640 (for this xkb, mi, fb and xfixes is sufficient).

If fb is really sufficient for this bug report to be closed, closing again...

**Comment by uli42** [comment]
Ah, right. That was only valid for the #641 patch series. Now that
they are all separated that statement is obsolete.

Uli
